### PR TITLE
Fix Loading an SVG with TextureLoader

### LINKED ISSUE #9575: Loading an SVG with TextureLoader no longer works as of r78.
##### Description of the problem

The following code works in R77, but as of R78, the `onLoad` callback will never be called, even when `onProgress` shows the file is 100% loaded. As I was unable to find any documentation about this change, I wanted to bring it to your attention as a possible bug.

```
var url = ; //any svg
new THREE.TextureLoader().load(url, () => {console.log("texture loaded");}, (xhr) => {console.log((xhr.loaded / xhr.total * 100) + '% loaded');});
```
##### Three.js version
- [x] Dev
- [x] r80
- [x] r79
- [x] r78
##### Browser
- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer
##### OS
- [x] All of them
- [ ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

### PR DESCRIPTION
**Pull request by kevanstannard**
The problem appears to be happening in the `ImageLoader` when converting the loaded SVG blob into a URL and then making it the src of an image:

~~~js
image.src = URL.createObjectURL( blob );
~~~

With the current code we override the server provided MIME type with `text/plain` (which ends up in `blob.type`).

For some reason creating PNGs and JPGs with a MIME type of `text/plain` works. Perhaps it's because it does not rely on the MIME type and instead derives it from the image binary content? Not sure, but with SVGs it results in an error.

This PR changes the behaviour so it only sets the overriding `text/plain` MIME type when not a blob. 

At the moment only `ImageLoader` requests a blob type so this PR should only affect loading of images. All other loaders should continue to use the `text/plain` MIME type.

Unfortunately I've been unable to reproduce errors/warnings reported by others in my dev environment, perhaps due to different local web servers, but hopefully this PR has a smaller impact that my previous PR (https://github.com/mrdoob/three.js/pull/10329)

As mentioned previously this should fix https://github.com/mrdoob/three.js/issues/9575

### CODE REVIEW

**Comment by mrdoob** [comment]
How about doing something like...

```js
if ( /\.svg$/.test( url ) ) this.mimeType = 'image/svg+xml';
```

**Comment by kevanstannard** [comment]
Honestly I was hoping to avoid a solution like that but it's probably the simplest and safest fix for now. Thanks for the suggestion, I'll update the PR.

It would be nice to see the FileLoader move away from this approach of overriding server provided MIME types. Forcing images and other types to all have a `text/plain` MIME type does feel like the wrong thing to do.

Do you have any ideas of what environments lead to the console warnings? If I can reproduce that then I have some other ideas that might be a better long term solution.

**Comment by Villmer** [comment]
Can't load an svg using TextureLoader or ImageLoader in version 83. Had to revert all the way back to version 76. :-(

**Comment by makc** [comment]
guys, I was the one who placed text/plain there. iirc, it was to make firefox shut up when loading files locally. since everyone is now using some crappy build system running local web server, this is not an issue at all, versus files not being loaded. feel free to remove that override.

**Comment by kevanstannard** [comment]
@makc thanks for chiming in.

If this was a fix to cater for loading files directly into the browser (i.e. using the `file://` protocol) then, as you've suggested, perhaps it should be removed now.

The change to use `XMLHttpRequest` for loading files means that the `file://` protocol will no longer work when loading files anyway (due to cross origin errors) - a web server is now required, so MIME types should always be present.

Although I am confused that @mrdumb reported seeing "Not well formed" console warnings for obj and js files. Not sure what's happening there.

@mrdoob what do you think? Would you be happy to remove this code? If warnings are reported then I would be happy to investigate with the reporter.

Also, it looks like we might need to explain in the docs "Introduction" that a web server is the preferred method of running three.js code.

**Comment by makc** [comment]
the obj files are essentially text, but xhr may be reading them as binary, since who would realistically configure text/plain for obj on web server? you probably need the override for those particular loaders.

**Comment by RemusMar** [comment]
> a web server is now required

Why should I need that (a security hole anyway) to test a simple JS or OBJ file?

**Comment by makc** [comment]
in theory you can still get away with [this](http://stackoverflow.com/a/4819114) but that's easier to give up and go with the flow on this one

**Comment by RemusMar** [comment]
> in theory you can still get away with this but that's easier to give up and go with the flow on this one

I use that setting from the very beginning.
But I did never like Chrome.
I use it only to test the compatibility of "something" on various browsers.
I use Firefox.

p.s.
http://localhost combined with a static / dedicated IP is an open door to hackers.
So use it only if it's a MUST !

File: src/loaders/FileLoader.js [hunk 1]
```diff
@@ -184,13 +184,12 @@ Object.assign( FileLoader.prototype, {
 			if ( this.responseType !== undefined ) request.responseType = this.responseType;
 			if ( this.withCredentials !== undefined ) request.withCredentials = this.withCredentials;
 
-			if ( ! this.mimeType && this.responseType !== 'blob' ) {
+			// Setting 'text/plain' as the MIME type for SVG files will cause a
+			// problem when the ImageLoader tries to use it as the src of an image,
+			// so explicitly set the SVG MIME type.
+			if ( /\.svg$/.test( url ) ) this.mimeType = 'image/svg+xml';
```
Comment (mrdoob) [inline src/loaders/FileLoader.js]: Actually, I think we should put this in `ImageLoader` instead.

Comment (kevanstannard) [inline src/loaders/FileLoader.js]: OK, will update.

Just wondering why you prefer not to remove the `text/plain` override?

**Comment by RemusMar** [comment]
> Actually, I think we should put this in ImageLoader instead.

+1

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**